sql/python/sqlflow_submitter/db.py

```diff
@@ -55,6 +56,64 @@ def execute(driver, conn, statement):
     return field_names, field_columns
 
 
+def db_generator(driver, conn, statement,
+                 feature_column_names, label_column_name,
+                 column_name_to_type, fetch_size=128):
+    def reader():
```

Comment:
It seems like this `reader` function and `execute` function looks similar, maybe we can combine them into a single function which can be controlled by argument to return all data or iterator.

Reply:
If the situation when `label_column_name` is null can be handled correctly, can we remove `db_generator_predict `?

Reply:
I've been thinking about this when I'm writing this PR, adding this type if conditional check may let this generator function a bit hard to understand, and inside the generator, adding runtime checking string compare also slow down the execution.